x/ibc/core/02-client/keeper/proposal.go

```diff
@@ -70,3 +70,35 @@ func (k Keeper) ClientUpdateProposal(ctx sdk.Context, p *types.ClientUpdatePropo
 
 	return nil
 }
+
+// UpgradeProposal sets the upgraded client state in the upgrade store. It clears
+// an IBC client state and consensus state if a previous plan was set. Then  it
+// will schedule an upgrade and finally set the upgraded client state in upgrade
+// store.
+func (k Keeper) UpgradeProposal(ctx sdk.Context, p *types.UpgradeProposal) error {
+	// clear any old IBC state stored by previous plan
+	oldPlan, exists := k.GetUpgradePlan(ctx)
+	if exists {
+		k.upgradeKeeper.ClearIBCState(ctx, oldPlan.Height-1)
+	}
```

Comment:
This should not exist here.

Reply:
The upgrade keeper should be in charge of ensuring there is only 1 plan active at a time. IBC keeper should assume this as invariant.